File: leetCode/Python2/258-addDigits.py
class Solution(object):
    def addDigitsJunior(self, num):
        """
        :type num: int
        :rtype: int
        """
        final_digit = num
        while final_digit > 9:
            final_digit = 0
            while num > 0:
                final_digit += num % 10
                num //= 10
            num = final_digit
        return final_digit

    def addDigits(self, num):
        if num == 0:
            return 0
        return num - 9 * ((num - 1) / 9)

if __name__ == "__main__":
    s = Solution()
    print s.addDigits(38)
    print s.addDigits(11)
    print s.addDigits(3)

File: leetCode/C++/258-addDigits.cc
class Solution1 { // iterative solution
public:
    int addDigits(int num) {
        int ans = 10;
        while (ans >= 10) {
            ans = 0;
            while (num) {
                ans += num % 10;
                num /= 10;
            }
            num = ans;
        }
        return ans;
    }
};

class Solution {
public:
    int addDigits(int num) {
        if (!num)
            return num;
        return num - 9 * ((num - 1) / 9);
    }
};